File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> id;
        int n=nums.size();
        for(int i=0;i<n;i++){
            id[nums[i]]=i;
        }
        for(int i=0;i<n;i++){
            int dif=target-nums[i];
            if(id.count(dif) && id[dif]!=i){
                return {i,id[dif]};
            }
        }
        return {};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, s: List[int], target: int) -> List[int]:
        n=len(s)
        l=0
        r=n-1
        sum=0
        while l<r:
            sum=s[l]+s[r]
            if sum==target:
                st=[l+1,r+1]
                return st
            elif sum>target:
                r-=1
            elif sum<target:
                l+=1


